lib/pipelineTemplateVersionFactory.js

```diff
@@ -50,90 +51,74 @@ class PipelineTemplateVersionFactory extends BaseFactory {
-                    if (!isExactVersion) {
-                        newVersion = config.version;
-                    } else if (!pipelineTemplate.latestVersion) {
-                        newVersion = configMinor ? `${configMajor}${configMinor}.0` : `${configMajor}.0.0`;
-                    } else {
-                        const [, globalMajor, globalMinor, globalPatch] = VERSION_REGEX.exec(
-                            pipelineTemplate.latestVersion
-                        );
-                        const patch = parseInt(globalPatch.slice(1), 10) + 1;
-
-                        newVersion = `${globalMajor}${globalMinor}.${patch}`;
-                    }
-                }
-
-                return super
-                    .create({
-                        templateId: pipelineTemplate.id,
-                        description: config.description,
-                        config: config.config,
-                        createTime,
-                        version: newVersion
-                    })
-                    .then(pipelineTemplateVersion => {
-                        return { pipelineTemplateVersion, pipelineTemplate };
-                    });
-            })
-            .then(result => {
-                const { pipelineTemplate, pipelineTemplateVersion } = result;
-                const [, latestMajor, latestMinor] = VERSION_REGEX.exec(pipelineTemplate.latestVersion);
-                const [, major, minor] = VERSION_REGEX.exec(pipelineTemplateVersion.version);
-
-                if (major > latestMajor || (major === latestMajor && minor >= latestMinor)) {
-                    pipelineTemplate.latestVersion = pipelineTemplateVersion.version;
-                    pipelineTemplate.updateTime = new Date().toISOString();
-                }
-
-                return pipelineTemplate.update().then(() => {
-                    return pipelineTemplateVersion;
-                });
+                maintainer: config.maintainer,
+                createTime,
+                updateTime: createTime
+            });
+        }
+
+        const exactVersionMatch = EXACT_VERSION_REGEX.exec(config.version);
+        let versionedConfig = null;
+
+        if (exactVersionMatch) {
+            versionedConfig = await this.get({
+                templateId: pipelineTemplate.id,
+                version: config.version
             });
+        }
+
+        const currentVersion = versionedConfig ? versionedConfig.version : config.version;
+        const [, configMajor, configMinor] = VERSION_REGEX.exec(currentVersion);
+        let newVersion;
+        let patch;
+
+        if (exactVersionMatch) {
+            newVersion = currentVersion;
+        } else if (!pipelineTemplate.latestVersion) {
+            newVersion = configMinor ? `${configMajor}${configMinor}.0` : `${configMajor}.0.0`;
+        } else {
+            const [, globalMajor, globalMinor, globalPatch] = VERSION_REGEX.exec(pipelineTemplate.latestVersion);
```

Comment:
i would have done something to parse out the numbers, but as long as you are parsing them correctly it's fine